<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>面试题</title>
</head>
<body>
    <script>
        // Promise.resolve()
        //     .then(() => {
        //         console.log(1);//正常执行
        //     })
        //     .catch(() => {
        //         console.log(2);//因为是成功状态，并没有失败状态，所以catch()方法不会执行
        //     })
        //     .then(() => {
        //         console.log(3);//跳过catch()继续执行
        //     })
        //     //最终控制台输出1 3


        // Promise.resolve()
        //     .then(() => {
        //         console.log(1);//1
        //         // 抛出异常，变为失败状态
        //         throw new Error("error1");
        //     })
        //     // 失败状态catch()继续执行
        //     .catch(() => {
        //         console.log(2);//2

        //     })
        //     // catch()默认 return undefined，所以then()继续执行
        //     .then(() => {
        //         console.log(3);//3
        //     });

        Promise.resolve()
            .then(() => {
                console.log(1);//1
                // 状态变为失败
                throw new Error("error1");
            })
            .catch(() => {
                // 抓住错误，执行catch()
                // 默认返回return undefined，成功状态
                console.log(2);//2
            })
            // 不执行
            .catch(() => {
                console.log(3);
            })
    </script>
</body>
</html>